### PR TITLE
Fix race conditions in preference bindings

### PR DESCRIPTION
**Pull request by fpabl0**
### Description:

Fixes race conditions in preference bindings.

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: data/binding/binding_test.go [hunk 1]
```diff
@@ -22,7 +22,6 @@ func TestBase_AddListener(t *testing.T) {
 	assert.Equal(t, 1, len(data.listeners))
 
 	waitForItems()
-	data.trigger()
```
Comment (andydotxyz) [inline data/binding/binding_test.go:25]: I’m curious about this change, can you help?

Comment (fpabl0) [reply data/binding/binding_test.go:25]: Yes, `data.AddListener(fn)` already enqueues DataChanged callback, so trigger it again is not needed, is it?

Comment (andydotxyz) [reply data/binding/binding_test.go:25]: That looks to be the case

**Review by andydotxyz — APPROVED** [review]

**Comment by andydotxyz** [comment]
I felt like I hit some hangs whilst testing this, but not certain this was the cause

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Freezes now reproducible it seems.
Open fyne_demo on non-data tab, then switch to data binding.
Sometimes hangs

**Comment by fpabl0** [comment]
> Open fyne_demo on non-data tab, then switch to data binding.
Sometimes hangs

Oh that is bad, I will investigate.

**Comment by fpabl0** [comment]
> Sometimes hangs

I am not able to reproduce it. does it happen frequently?

**Comment by fpabl0** [comment]
With this last commit, do you still experiment freeze?

**Comment by andydotxyz** [comment]
> With this last commit, do you still experiment freeze?

I can't seem to replicate it any more.

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**